### PR TITLE
Implemented useScrollToTop

### PR DESCRIPTION
**Pull request by frankcalise**
## Please verify the following:

- [ ] `yarn ci:test` **jest** tests pass with new tests, if relevant

## Describe your PR

Implements request from #1810 - Add #1810 useScrollToTop in Screen component. Pressing the active tab will scroll to the top of the content within the `<ScrollView />`.

scroll-to-top

### CODE REVIEW

File: boilerplate/app/components/Screen.tsx [hunk 1]
```diff
@@ -159,12 +160,17 @@ function ScreenWithScrolling(props: ScreenProps) {
...
 
+  // Add native behavior of pressing the active tab to scroll to the top of the content
+  // More info at: https://reactnavigation.org/docs/use-scroll-to-top/
+  useScrollToTop(ref)
+
   return (
     <ScrollView
-      keyboardShouldPersistTaps={keyboardShouldPersistTaps}
-      scrollEnabled={scrollEnabled}
+      {...{ keyboardShouldPersistTaps, scrollEnabled, ref }}
```
Comment (frankcalise) [inline boilerplate/app/components/Screen.tsx:173]: I'm fine with reverting this back, but for a bunch of props that just pass the same name/var the shorthand can look less busy when reading through the render function.

Comment (yulolimum) [reply boilerplate/app/components/Screen.tsx:173]: I actually kinda like this. Will try using more.

File: boilerplate/app/components/Screen.tsx [hunk 2]
```diff
@@ -159,12 +160,17 @@ function ScreenWithScrolling(props: ScreenProps) {
     style,
   } = props as ScrollScreenProps
 
+  const ref = useRef<ScrollView>()
+
   const { scrollEnabled, onContentSizeChange, onLayout } = useAutoPreset(props as AutoScreenProps)
 
+  // Add native behavior of pressing the active tab to scroll to the top of the content
+  // More info at: https://reactnavigation.org/docs/use-scroll-to-top/
+  useScrollToTop(ref)
```
Comment (yulolimum) [inline boilerplate/app/components/Screen.tsx:169]: Do you know, is this a no-op if the screen is inside a stack or a nav other than tab?

Comment (frankcalise) [reply boilerplate/app/components/Screen.tsx:169]: @yulolimum `react-navigation` early returns from the `useEffect` if there is no tab nav found, so no listener would be set up in that case.

image

See this file for more info:
https://github.com/react-navigation/react-navigation/blob/main/packages/native/src/useScrollToTop.tsx#L56

**Review by jamonholmgren — APPROVED** [review]

**Pull request merged**